### PR TITLE
Optimize `convertToLocalDateTime` method

### PR DESCRIPTION
**Pull request by leonchen83**
Signed-off-by: Baoyi Chen <chen.bao.yi@qq.com>
Original PR: #456

### CODE REVIEW

**Comment by chrjohn** [comment]
@leonchen83 
Could you remove the leap second fix from this PR and open a separate with the changes from this (i.e. #529) PR plus the leap second fix against master?
Or alternatively open another PR *only* with the optimization but *without* the leap second fix against master.
Thank you.

**Comment by leonchen83** [comment]
sorry reply late. I rollbacked leap second code. please merge this PR first

File: quickfixj-core/src/main/java/quickfix/field/converter/UtcTimestampConverter.java [hunk 1]
```diff
@@ -135,26 +135,37 @@ public static Date convert(String value) throws FieldConvertError {
...
             }
-        } catch (DateTimeParseException e) {
+    
+            int yy = parseInt(value, 0, 4);
+            int mm = parseInt(value, 4, 2);
+            int dd = parseInt(value, 6, 2);
+            int h = parseInt(value, 9, 2);
+            int m = parseInt(value, 12, 2);
+            int s = parseInt(value, 15, 2);
+            return LocalDateTime.of(yy, mm, dd, h, m, s, ns);
```
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/field/converter/UtcTimestampConverter.java:155]: Probably a micro optimisation, but could we have only the `LocalDateTime.of` in the try-block? It shouldn't be necessary for the `parseInt()` method.

Comment (leonchen83) [reply quickfixj-core/src/main/java/quickfix/field/converter/UtcTimestampConverter.java:155]: that make sense

**Comment by chrjohn** [comment]
Hi @leonchen83 ,
thanks for the PR. I have just merged it. Before you create a PR for the leap second stuff, please wait until further notice. I need to clean something up first.
Cheers,
Chris.

**Pull request merged**